src/meta-srv/src/election/etcd.rs

```
@@ -297,3 +296,48 @@ impl Election for EtcdElection {
         self.leader_watcher.subscribe()
     }
 }
+
+impl EtcdElection {
+    async fn keep_alive(
+        &self,
+        keeper: &mut LeaseKeeper,
+        receiver: &mut LeaseKeepAliveStream,
+        leader: &LeaderKey,
+    ) -> Result<()> {
+        keeper.keep_alive().await.context(error::EtcdFailedSnafu)?;
+        if let Some(res) = receiver.message().await.context(error::EtcdFailedSnafu)? {
+            if res.ttl() > 0 {
+                // Only after a successful `keep_alive` is the leader considered official.
+                if self
+                    .is_leader
+                    .compare_exchange(false, true, Ordering::Relaxed, Ordering::Relaxed)
+                    .is_ok()
+                {
+                    self.infancy.store(true, Ordering::Relaxed);
+
+                    if let Err(e) = self
+                        .leader_watcher
+                        .send(LeaderChangeMessage::Elected(Arc::new(leader.clone())))
+                    {
+                        error!(e; "Failed to send leader change message");
+                    }
+                }
+            } else {
+                if self.is_leader.load(Ordering::Relaxed) {
+                    if let Err(e) = self
+                        .leader_watcher
+                        .send(LeaderChangeMessage::StepDown(Arc::new(leader.clone())))
+                    {
+                        error!(e; "Failed to send leader change message");
+                    }
+                }
```

Let's just return the error, and handle the Stepdown outside